src/scvi/dataloaders/_data_splitting.py

```diff
@@ -34,27 +38,77 @@ def validate_data_split(
         Size of train set. Need to be: 0 < train_size <= 1.
     validation_size
         Size of validation set. Need to be 0 <= validation_size < 1
+    use_external_indexing
+        Whether to use external supported indexing. This bypass any other flag or input parameter
+        that was before
+    external_indexing
+        A list of np.arrays that is always in the order of [[train_idx],[valid_idx],[test_idx]].
     """
-    if train_size > 1.0 or train_size <= 0.0:
-        raise ValueError("Invalid train_size. Must be: 0 < train_size <= 1")
+    if use_external_indexing:
+        if external_indexing is None:
+            raise ValueError("External indexing list is missing")
+
+        if type(external_indexing) is not list:
+            raise ValueError("External indexing is not of list type")
+
+        # validate the structure of it
+        if type(external_indexing[0]) is not np.ndarray:
+            raise ValueError("The first element of the list of lists is not a np.array")
```

Comment:
For the sake of not repeating these checks explicitly for each element, you can do something like:
```
if not all(isinstance(elem, np.ndarray) for elem in external_indexing)):
   # raise error here
```
No need to specify which element is not the correct type - I think the users can figure that out.

Reply:
Sure, I actually added here handling for the case where the 2nd or 3rd arrays are even not given to us, and imputed that . But made it now more compact.